### PR TITLE
AuthZ: Use new PolicyEvaluator

### PR DESCRIPTION
**Pull request by HaoK**
@BrennanConroy 

Depends on https://github.com/aspnet/Security/pull/1193

### CODE REVIEW

**Comment by BrennanConroy** [comment]
Looks good, should we add a test or 2 for forbids?

**Comment by davidfowl** [comment]
Why are the builds still failing?

**Comment by BrennanConroy** [comment]
> Depends on aspnet/Security#1193

Which hasn't been merged

File: src/Microsoft.AspNetCore.SignalR.Http/SignalRDependencyInjectionExtensions.cs [hunk 1]
```diff
@@ -10,6 +10,7 @@ public static class SignalRDependencyInjectionExtensions
         public static ISignalRBuilder AddSignalR(this IServiceCollection services)
         {
             services.AddSockets();
+            services.AddAuthorizationPolicyEvaluator();
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.SignalR.Http/SignalRDependencyInjectionExtensions.cs]: This is in the wrong place. It should be in AddSockets not AddSignalR

**Comment by davidfowl** [comment]
The dev branch still looks red to me. Do we just need to force a rebuild?

**Comment by HaoK** [comment]
Yeah individual repos broken by other repos need to be manually triggered

**Comment by HaoK** [comment]
I'll add a few forbid tests and move the Add to the right place

**Comment by HaoK** [comment]
Updated PR, looks like the tests needed to be switched over to Auth 2.0 as well

**Comment by moozzyk** [comment]
@HaoK - I can no longer build SignalR repo. The build fails with: Unable to resolve 'Microsoft.AspNetCore.Authorization.Policy (>= 2.0.0)' for '.NETCoreApp,Version=v2.0'. [C:\source\SignalR-Core\.build\KoreBuild.proj]. The package is missing from aspnetcore-ci-dev.

**Comment by HaoK** [comment]
Yup, still working on the universe coherence build, was a 11 repo change, so I wouldn't expect packages til later tonight

**Comment by HaoK** [comment]
If you want to work against the volitale packages so you can work against the new policy, you can add <add key="AspNetCore" value="https://dotnet.myget.org/F/aspnetcore-volatile-dev/api/v3/index.json" /> to your nuget.config

**Comment by moozzyk** [comment]
Thanks! Will try the volatile feed.

File: test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs [hunk 2]
```diff
@@ -693,31 +691,23 @@ public class HttpConnectionDispatcherTests
...
+            // would hang if EndPoint was running
+            await dispatcher.ExecuteAsync(context, options, app).OrTimeout();
 
-            Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
-            Assert.Equal("T12:T:Hello, World;", GetContentAsString(context.Response.Body));
+            Assert.Equal(StatusCodes.Status403Forbidden, context.Response.StatusCode);
         }
 
         [Fact]
-        public async Task AllPoliciesRequiredForAuthorizedEndPoint()
```
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs]: Why did you remove this test?

Comment (HaoK) [inline test/Microsoft.AspNetCore.Sockets.Tests/HttpConnectionDispatcherTests.cs]: Oops, I used it as the basis for the new forbid test, but must have deleted the original accidentally when I did the conversion to the new auth 2.0 stuff.
